tests/js/server/dump/dump-setup-mixed.js

```diff
@@ -0,0 +1,309 @@
+    }
+  }
+
+  c = db._createEdgeCollection("UnitTestsDumpEdges");
+  for (i = 0; i < 10; ++i) {
+    c.save("UnitTestsDumpMany/test" + i,
+           "UnitTestsDumpMany/test" + (i + 1),
+           { _key: "test" + i, what: i + "->" + (i + 1) });
+  }
+
+  // we update & modify the order
+  c = db._create("UnitTestsDumpOrder");
+  c.save({ _key: "one", value: 1 });
+  c.save({ _key: "two", value: 2 });
+  c.save({ _key: "three", value: 3 });
+  c.save({ _key: "four", value: 4 });
+
+  c.update("three", { value: 3, value2: 123 });
+  c.update("two", { value: 2, value2: 456 });
+  c.update("one", { value: 1, value2: 789 });
+  c.remove("four");
+
+  // we apply a series of updates & removals here
+  c = db._create("UnitTestsDumpRemoved");
+  c.save(Array(10000).fill().map((e, i, a) => Object({_key: "test" + i, value1: i})));
+  for (i = 0; i < 1000; ++i) {
+    c.update("test" + i, { value2: i + 1 });
+  }
+  c.remove(Array(1000).fill().map((e, i, a) => "test" + (i * 10)));
+
+  // we create a lot of (meaningless) indexes here
+  c = db._create("UnitTestsDumpIndexes");
+  c.ensureUniqueConstraint("a_uc");
+  c.ensureSkiplist("a_s1", "a_s2");
+
+  c.ensureHashIndex("a_h1", "a_h2");
+  c.ensureUniqueSkiplist("a_su");
+  c.ensureHashIndex("a_hs1", "a_hs2", { sparse: true });
+  c.ensureSkiplist("a_ss1", "a_ss2", { sparse: true });
+  c.ensureFulltextIndex("a_f");
+
+  c.ensureGeoIndex("a_la", "a_lo");
+
+  // we insert data and remove it
+  c = db._create("UnitTestsDumpTruncated");
+  c.save(Array(10000).fill().map((e, i, a) => Object({_key: "test" + i, value1: i, value2: "this is a test", value3: "test" + i})));  
+  c.truncate({ compact: false });
+
+  //// custom key options
+  //c = db._create("UnitTestsDumpKeygen", {
+  //  keyOptions: {
+  //    type: "autoincrement",
+  //    allowUserKeys: false,
+  //    offset: 7,
+  //    increment: 42
+  //  }
+  //});
+  //docs = [];
+  //for (i = 0; i < 1000; ++i) {
+  //  docs.push({ value: i, more: { value: [ i, i ] } });
+  //}
+  //c.save(docs);
+  //let d = c.save({});
+  //c.save({}); // create another one inbetween...
+  //c.remove(d);
+
+  // custom key options
+  c = db._create("UnitTestsDumpKeygenPadded", {
+    keyOptions: {
+      type: "padded",
+      allowUserKeys: false
+    }
+  });
+
+  docs = [];
+  for (i = 0; i < 1000; ++i) {
+    docs.push({ value: i, more: { value: [ i, i ] } });
+  }
+  c.save(docs);
+  let d = c.save({});
+  c.save({}); // create another one inbetween...
+  c.remove(d);
+  process.env['paddedLastValue'] = c.properties().keyOptions.lastValue;
+  
+  // custom key options
+  c = db._create("UnitTestsDumpKeygenUuid", {
+    keyOptions: {
+      type: "uuid",
+      allowUserKeys: false
+    }
+  });
+  docs = [];
+  for (i = 0; i < 1000; ++i) {
+    docs.push({ value: i });
+  }
+  c.save(docs);
+  d = c.save({});
+  c.save({}); // create another one inbetween...
+  c.remove(d);
+
+  // strings
+  c = db._create("UnitTestsDumpStrings");
+  var texts = [
```

Comment:
This file seems to be in general a copy/paste. Approx 90% here is unrelated to the keygen tests. That texts string I've seen at least twice. Please export to a helper.